learning/tour-of-beam/frontend/lib/pages/tour/widgets/unit_content.dart

```diff
@@ -30,6 +32,76 @@ class UnitContentWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MarkdownBody(data: unitContent.description);
+    final textTheme = Theme.of(context).textTheme;
+    final markdownStyle = MarkdownStyleSheet(
+      h1: textTheme.headlineLarge,
+      h3: textTheme.headlineMedium,
+      p: textTheme.bodyMedium,
+      code: Theme.of(context)
+          .extension<BeamThemeExtension>()!
+          .codeRootStyle
+          .copyWith(
+            backgroundColor: BeamColors.transparent,
+            fontSize: BeamSizes.size12,
+          ),
+      codeblockDecoration: BoxDecoration(
+        // TODO(nausharipov): where to store these colors?
+        color: Theme.of(context).brightness == Brightness.dark
+            ? BeamDarkThemeColors.codeBackground
+            : BeamLightThemeColors.codeBackground,
+        border: Border.all(color: Theme.of(context).primaryColor),
+        borderRadius: const BorderRadius.all(
+          Radius.circular(BeamSizes.size4),
+        ),
+      ),
+    );
+
+    return Markdown(
+      selectable: true,
+      data: unitContent.description,
+      builders: {
+        'code': CodeBuilder(),
+      },
+      styleSheet: markdownStyle,
+    );
+  }
+}
+
+class CodeBuilder extends MarkdownElementBuilder {
+  @override
+  Widget? visitElementAfter(md.Element element, TextStyle? preferredStyle) {
+    final String textContent = element.textContent;
+    final bool isMultilineCode = textContent.contains('\n');
+    if (isMultilineCode) {
+      /// codeblockDecoration is applied for multiline code
+      return null;
+    }
+    return _InlineCode(text: textContent);
+  }
+}
+
+// TODO(nausharipov): is inline the right word?
+class _InlineCode extends StatelessWidget {
+  final String text;
+  const _InlineCode({required this.text});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(
+        horizontal: BeamSizes.size3,
+        vertical: BeamSizes.size1,
+      ),
+      decoration: BoxDecoration(
+        color: Theme.of(context).brightness == Brightness.dark
```

Comment:
Add `BeamThemeExtension.codeBackgroundColor` instead of checking brightness here.